lib/ProductOpener/Tags.pm

```diff
@@ -2125,13 +2172,20 @@ sub build_tags_taxonomy ($tagtype, $publish) {
 
 		close $OUT;
 
-		if ($errors ne "") {
+		if (scalar @taxonomy_errors) {
 
 			print STDERR "Errors in the $tagtype taxonomy definition:\n";
-			print STDERR $errors;
+			print STDERR join("", map {_taxonomy_error_display($_)} @taxonomy_errors);
+			# do we only have duplicate synonyms errors ?
+			my $only_duplicate_errors = !(first {%{$_}{type} ne "duplicate_synonym"} @taxonomy_errors);
 			# Disable die for the ingredients taxonomy that is merged with additives, minerals etc.
-			# Disable also for packaging taxonomy for some shapes and materials shares same names
-			unless (($tagtype eq "ingredients") or ($tagtype eq "packaging") or ($tagtype eq "inci_functions")) {
+			# Disable die for the packaging taxonomy as some legit material and shape might have same name
+			my $taxonomy_with_duplicate_tolerated
+				= (($tagtype eq "ingredients") or ($tagtype eq "packaging") or ($tagtype eq "inci_functions"));
+			# FIXME: special excetion for ingredients taxonomy until we fix errors !
+			$only_duplicate_errors = $only_duplicate_errors || ($tagtype eq "ingredients");
```

Comment:
we should remove this after merging https://github.com/openfoodfacts/openfoodfacts-server/pull/10052

Reply:
```suggestion
```